src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/loadbalancer/ApacheEc2LoadBalancerFormula.kt

```diff
@@ -13,7 +14,9 @@ import org.apache.logging.log4j.Logger
 import java.net.URI
 import java.time.Duration
 
-internal class ApacheEc2LoadBalancerFormula : LoadBalancerFormula {
+class ApacheEc2LoadBalancerFormula : LoadBalancerFormula {
+
+    private var uri: URI? = null
```

Comment:
This mutability is a smell. There's a hidden contract between `provision` and `updateJiraConfiguration`. One must be called after another.
And it is bad for thread-safety if you reuse the same LB formula for two cohorts.

Perhaps `updateJiraConfiguration` should be part of `LoadBalancer` API.

Reply:
I'll add `updateJiraConfiguration` to part of `ProvisionedLoadBalancer`.

Reply:
Could the `LoadBalancer` have it? This way we could empower all SSH-based use cases, not only AWS-SSH ones. We could move `HaProxyLoadBalancer` to `infrastructure`, which has a wider audience and a more lightweight dev loop.

Reply:
Done.